docs/testing.md

```diff
@@ -0,0 +1,91 @@
+# Testing considerations
+
+This document contains information about how the tests are structured and why.
+
+## Types of test
+
+There are two types of tests:
+
+- Unit tests, which run tests on classes or subsets of the application in a single JVM. Those run with `./gradlew test`.
+- Integration tests, which launch the UAA application and run web-based tests the running app. Those can be run with
+  `./gradlew integrationTest`
+
+There are helper scripts, `run-unit-tests.sh` and `run-integration-tests.sh`, which run the tests inside a docker
+container. The docker container they run in contains the database against which to run the tests, as well as an LDAP
+server.
+
+## Test databases
+
+By default, the tests run against an in-memory DB, `hsqldb`. This DB is also present in the prod artifact, so that
+UAA can also be ran standalone to test tweaks in a live instance.
+
+To run these databases locally, use the docker-compose script:
+
+    $ docker compose --file scripts/docker-compose.yaml up
+
+If you wish to launch only one of the DBs, select the appropriate service name:
+
+    $ docker compose --file scripts/docker-compose.yaml up postgresql
+
+To run tests against either Postgres or MySQL, use the `postgresql` or `mysql` profile, to select the DB. Be sure
+to add the `default` profile which will trigger seeding the database with some admin users, clients, etc. For example:
+
```

Comment:
in main doc we state to test with 
https://github.com/cloudfoundry/uaa/blob/develop/run-integration-tests.sh
and
https://github.com/cloudfoundry/uaa/blob/develop/run-unit-tests.sh

Reply:
Amended to reflect this